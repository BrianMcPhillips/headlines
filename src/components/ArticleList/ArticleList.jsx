import React from 'react';
import PropTypes from 'prop-types';
import Article from '../Article/Article';

const ArticleList = ({ data }) => {
  const articleElements = data.map(article => (
    <li key={article.url}>
      <Article {...article}/>
    </li>
  ));
  return (
    <ul data-testid="articles">
      {articleElements }
    </ul>
  );
};

ArticleList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    urlToImage: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    author: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default ArticleList;
