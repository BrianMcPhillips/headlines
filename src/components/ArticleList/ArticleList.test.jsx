import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ArticleList from './ArticleList';

describe('ArticleList component', () => {
  afterEach(() => cleanup());
  it('renders ArticleList', () => {
    const { asFragment } = render(<ArticleList data={[
      {
        urlToImage: 'http://placecage.com/200/200',
        title: 'Best Article',
        description: 'The best article',
        author: 'Best Article Writer',
        url: 'http://bestarticle.com'
      }
    ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
