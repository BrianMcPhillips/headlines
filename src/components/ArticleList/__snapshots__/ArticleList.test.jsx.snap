// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList component renders ArticleList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <div>
        <img
          alt="Best Article"
          src="http://placecage.com/200/200"
        />
        <h2>
          Best Article
        </h2>
        <h4>
          The best article
        </h4>
        <p>
          Best Article Writer
        </p>
        <a
          href="http://bestarticle.com"
        >
          View Article
        </a>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
